### PR TITLE
[Eigen] Build for Experimental Platforms

### PR DESCRIPTION
**Pull request by Wimmerer**
Let's see what happens. I got `LoadError: Unable to automatically install 'GCCBootstrap-x86_64-linux-musl.v10.2.0.x86_64-linux-musl.unpacked` so let's see what this does. 

Also is empty products intentional?

### CODE REVIEW

**Comment by Wimmerer** [comment]
So, do what happens to `Lints.jl` if this gets merged?

File: E/Eigen/build_tarballs.jl [hunk 1]
```diff
@@ -27,7 +27,7 @@ install_license COPYING.*
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
-platforms = supported_platforms()
+platforms = supported_platforms(; experimental=true)
```
Comment (giordano) [inline E/Eigen/build_tarballs.jl:30]: No, there is no need to do this
```suggestion
platforms = [AnyPlatform()]
```

File: E/Eigen/build_tarballs.jl [hunk 2]
```diff
@@ -39,4 +39,4 @@ dependencies = Dependency[
 ]
 
 # Build the tarballs, and possibly a `build.jl` as well.
-build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies)
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6")
```
Comment (giordano) [inline E/Eigen/build_tarballs.jl:42]: No need to require Julia v1.6 for AnyPlatform. This package shouldn't even be instantiated by users at all, it's just header files 
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies)
```

**Review by giordano — CHANGES_REQUESTED** [review]

**Comment by Wimmerer** [comment]
Not sure why it still shows changes requested...

**Comment by giordano** [comment]
> Also is empty products intentional?

Yes, it's written in the comments: eigen is headers-only

> Not sure why it still shows changes requested...

Because GitHub doesn't track the changes

**Review by giordano — APPROVED** [review]

**Pull request merged**